README.md

```diff
@@ -30,24 +34,42 @@ This team is spun off from the [Modules team](https://github.com/nodejs/modules)
-   
-   - [ ] Follow up with similar helper functions that make up what happens within Node’s internal `load`. (Definitions to come.)
+   - [ ] Convert `resolve` from async to sync https://github.com/nodejs/node/pull/43363
+
+      - [ ] Add an async `resolve` to [`module`](https://nodejs.org/api/module.html) module
+
+      - [ ] Consider an API for async operations before resolution begins, such as `preImport` https://github.com/nodejs/loaders/pull/89
+
+- [ ] Provide a way to register loaders without a command-line flag, for example via a `"loaders"` field in `package.json`.
 
 - [ ] Support loading source when the return value of `load` has `format: 'commonjs'`. See https://github.com/nodejs/node/issues/34753#issuecomment-735921348 and https://github.com/nodejs/loaders-test/blob/835506a638c6002c1b2d42ab7137db3e7eda53fa/coffeescript-loader/loader.js#L45-L50.
 
+- [ ] Finish and stabilize `--experimental-vm-modules` so that `vm` fully supports ESM.
+
+### Milestone 2: Usability improvements
+
 - [ ] First-class support for [import maps](https://github.com/WICG/import-maps) that doesn’t require a custom loader.
 
-We hope that moving loaders off thread will allow us to preserve an async `resolve` hook while supporting the sync `import.meta.resolve` API. If that turns out to be unachievable, however, then:
+- [ ] Add helper/utility functions to reduce boilerplate in user-defined hooks
+
+   - [ ] Start with the functions that make up the ESM resolution algorithm as defined in the [spec](https://nodejs.org/api/esm.html#resolver-algorithm-specification). Create helper functions for each of the functions defined in that psuedocode: `esmResolve`, `packageImportsResolve`, `packageResolve`, `esmFileFormat`, `packageSelfResolve`, `readPackageJson`, `packageExportsResolve`, `lookupPackageScope`, `packageTargetResolve`, `packageImportsExportsResolve`, `patternKeyCompare`. (Not necessarily all with these exact names, but corresponding to these functions from the spec.)
+
+   - [ ] Follow up with similar helper functions that make up what happens within Node’s internal `load`. (Definitions to come.)
+
+- [ ] Helper/utility functions to allow access to the CommonJS named exports discovery algorithm (`cjs-module-lexer`).
+
+- [ ] Hooks for customizing the REPL, including transpilation and tab completion. Support users pasting TypeScript (or CoffeeScript or whatever) into the REPL and having just as good an experience as with plain JavaScript.
+
+   - [ ] Support top-level `await` in the REPL, if possible.
```

Comment:
Node.js REPL already supports TLA, I think this point need to be rephrased to be more explicit on what's missing (or needs to be ticked off if we're happy with the current state of things).

Reply:
@cspotcode ?

Reply:
None of that stuff's exposed in the REPL API.  It's the class of issue you hit when trying to have first-class TS support.  E.g. ts-node lets you open a REPL and do `const a: string = await someApiCall(body) as const;` (contrived example)  But if anyone else tried to implement what we've done, they'd realize it's not fun.

The same happens with the mechanism where node guesses ESM named exports from CJS modules.  The named exports issue has been hit by the yarn folks, so if that one is elsewhere on this checklist, they're good people to talk to.